### ISSUE TITLE
base64 encoded attachment missing

### ISSUE THREAD

**Issue by rep-its** [jstedfast/MailKit#513]
Hi, 

Recently, i've stumpled over an email, which has an attachment when viewed im Microsoft Outlook 2013, but doesn't show it when I iterate through the MimeMessage.Attachment collection. 

`For Each attachment As MimeEntity In Email.Attachments`

However, the missing attachment shows up, when I iterate through all MimeEntities of the email. 
`For Each attachment As MimeEntity In Email.BodyParts.OfType(Of MimeEntity)()`

`Content-Type: application/octet-stream;
	name="Demo_PDF.pdf"
Content-Transfer-Encoding: base64`

I've saved the email with Mailkit to an .eml file and have it added here. 
[Mail.zip](https://github.com/jstedfast/MailKit/files/1014242/Mail.zip)

Note: I've changed all sensitive customer data in this email.

Edit: I'm using version 1.16.1 of Mimekit and Mailkit on .Net Framework 4.5.1.

Peter

PS: Great Framework!

**Comment by jstedfast** [comment]
The Attachments property only includes entities that have a Content-Disposition header with a value of "attachment ". The FAQ has more information about this.

**Comment by gluszczek** [comment]
Hi, I have a problem a little related to this.
The attachments (images) which are put to message content don't appear in 'Attachments' collection.
I send email from gmail to gmail client and put 1 attachment as additional file and 1 attachment as image in content, the result is that Attachments collection contains only this additional file. I use the same code like this: http://www.mimekit.net/docs/html/P_MailKit_IMessageSummary_Attachments.htm .
MailKit version 1.18, but also tried with 1.16.1.

I have noticed that I can detect all attachments using BodyParts iteration and check if BodyPartBasic contains "FileName". When I catch this attachment its properties are like this:
![image](https://user-images.githubusercontent.com/18527635/29840717-3548faae-8d03-11e7-8f99-db46a72f5624.png)
Entity type of its is MimePart. What is strange for me - the IsAttachment flag suggests 'false' and as I consider this is the reason why it is not in Attachments collection.

And I have also one more question. When I got the attachment using BodyParts iteration, how can I find out the location of this "image" in message content? To put it between text in the future view.

Thanks for help!
